src/util.c

```diff
@@ -126,9 +129,7 @@ int stringmatchlen(const char *pattern, int patternLen,
                 pattern++;
                 patternLen--;
             }
-            if (not)
-                match = !match;
-            if (!match)
+            if ((match && not) || (!match && !not))
```

Comment:
Would work for bool `if (match == not)`

Reply:
Thank you.
In the first, I want to use [match] to show why exit the while loop.
Then find this is no use to set match other than 1/0